authz/providers/azure/rbac/rbac.go

```diff
@@ -0,0 +1,259 @@
+		tokenProvider:            tokenProvider,
+		azureResourceId:          resourceId,
+		armCallLimit:             armCallLimit,
+		dataStore:                dataStore,
+		retrieveGroupMemberships: retrieveGroupMemberships,
+		skipAuthzForNonAADUsers:  skipAuthzForNonAADUsers}
+
+	u.skipCheck = make(map[string]void, len(skipList))
+	var member void
+	for _, s := range skipList {
+		u.skipCheck[strings.ToLower(s)] = member
+	}
+
+	if clsuterType == "arc" {
+		u.clusterType = connectedClusters
+	}
+
+	if clsuterType == "aks" {
+		u.clusterType = managedClusters
+	}
+
+	return u, nil
+}
+
+func New(clientID, clientSecret, tenantID, aadEndpoint, armEndPoint, clusterType, resourceId string, armCallLimit int, dataStore authz.Store, skipCheck []string, retrieveGroupMemberships, skipAuthzForNonAADUsers bool) (*AccessInfo, error) {
+	rbacURL, err := url.Parse(armEndPoint)
+
+	if err != nil {
+		return nil, err
+	}
+
+	tokenProvider := graph.NewClientCredentialTokenProvider(clientID, clientSecret,
+		fmt.Sprintf("%s%s/oauth2/v2.0/token", aadEndpoint, tenantID),
+		fmt.Sprintf("%s.default", armEndPoint))
+
+	return newAccessInfo(tokenProvider, rbacURL, clusterType, resourceId, armCallLimit, dataStore, skipCheck, retrieveGroupMemberships, skipAuthzForNonAADUsers)
+}
+
+func NewWithAKS(tokenURL, tenantID, armEndPoint, clusterType, resourceId string, armCallLimit int, dataStore authz.Store, skipCheck []string, retrieveGroupMemberships, skipAuthzForNonAADUsers bool) (*AccessInfo, error) {
+	rbacURL, err := url.Parse(armEndPoint)
+
+	if err != nil {
+		return nil, err
+	}
+	tokenProvider := graph.NewAKSTokenProvider(tokenURL, tenantID)
+
+	return newAccessInfo(tokenProvider, rbacURL, clusterType, resourceId, armCallLimit, dataStore, skipCheck, retrieveGroupMemberships, skipAuthzForNonAADUsers)
+}
+
+func (a *AccessInfo) RefreshToken() error {
+	resp, err := a.tokenProvider.Acquire("")
+	if err != nil {
+		glog.Errorf("%s failed to refresh token : %s", a.tokenProvider.Name(), err.Error())
+		return errors.Wrap(err, "failed to refresh rbac token")
+	}
+
+	// Set the authorization headers for future requests
+	a.headers.Set("Authorization", fmt.Sprintf("Bearer %s", resp.Token))
+	expIn := time.Duration(resp.Expires) * time.Second
+	a.expiresAt = time.Now().Add(expIn - expiryDelta)
+
+	return nil
+}
+
+func (a *AccessInfo) IsTokenExpired() bool {
+	if a.expiresAt.Before(time.Now()) {
+		return true
+	} else {
+		return false
+	}
+}
+
+func (a *AccessInfo) ShouldSkipAuthzCheckForNonAADUsers() bool {
+	return a.skipAuthzForNonAADUsers
+}
+
+func (a *AccessInfo) GetResultFromCache(request *authzv1.SubjectAccessReviewSpec) (bool, bool) {
+	var result bool
+	key := getResultCacheKey(request)
+	glog.V(10).Infof("Cache search for key: %s", key)
+	found, _ := a.dataStore.Get(key, &result)
+	return found, result
+}
+
+func (a *AccessInfo) SkipAuthzCheck(request *authzv1.SubjectAccessReviewSpec) bool {
+	if a.clusterType == connectedClusters {
```

Comment:
I am confused by this check. To me it seems that a.clusterType can be either "arc" or "aks"